sklearn/neighbors/graph.py

```diff
@@ -180,5 +186,275 @@ def radius_neighbors_graph(X, radius, mode='connectivity', metric='minkowski',
     else:
         _check_params(X, metric, p, metric_params)
 
-    query = _query_include_self(X, include_self)
+    query = _query_include_self(X._fit_X, include_self, mode)
     return X.radius_neighbors_graph(query, radius, mode)
+
+
+class KNeighborsTransformer(NeighborsBase, KNeighborsMixin,
+                            UnsupervisedMixin, TransformerMixin):
+    """Transform X into a (weighted) graph of k nearest neighbors
+
+    The transformed data is a sparse graph as returned by kneighbors_graph.
+
+    .. versionadded:: 0.21
+
+    Parameters
+    ----------
+    mode : {'distance', 'connectivity'}, optional (default = 'distance')
+        Type of returned matrix: 'connectivity' will return the connectivity
+        matrix with ones and zeros, and 'distance' will return the distances
+        between neighbors according to the given metric.
+
+    n_neighbors : int, optional (default = 5)
+        Number of neighbors to use for :meth:`kneighbors` queries.
```

Comment:
Specify that this includes the query point, so one should usually use `n_neighbors + 1`! This probably needs to be written quite boldly. I'm also concerned that it then makes little sense to use the same n_neighbors for `fit_transform` as for `transform`....